backend/src/main/java/com/jobwait/control/RequestController.java

```diff
@@ -8,35 +8,35 @@
 import com.jobwait.persistence.PersistenceController;
 import com.jobwait.persistence.PostgresController;
 import com.jobwait.security.AuthToken;
-import com.jobwait.security.LinkedInOAuthValidator;
-import com.jobwait.security.OAuthValidator;
+import com.jobwait.security.exchange.LinkedInOAuthExchange;
+import com.jobwait.util.Tuple;
 import com.jobwait.stat.StatRefreshController;
 import com.jobwait.stat.Stats;
 
 public class RequestController {
     private PersistenceController persistence = new PostgresController();
-    private OAuthValidator oAuthValidator = new LinkedInOAuthValidator();
     private StatRefreshController statRefreshController = new StatRefreshController();
+    private LinkedInOAuthExchange linkedInOAuthExchange = new LinkedInOAuthExchange();
 
-    public User getUserFromAuthToken(AuthToken token) {
-        oAuthValidator.validateToken(token);
-        User user = persistence.getUserFromAuthId(token.clientId());
+    public User getUserFromAuthCode(String authCode) {
+        String authHash = linkedInOAuthExchange.getUserIDAndHash(authCode).getRight();
```

Comment:
> getRight()

beloved monads? could it really be you after all this time?

Reply:
Realistically we should just call `Tuple` an `Either` because Tuples should not be limited to x elements so getRight() and getLeft() would not really make sense but whatever this works and codes alright too.

Reply:
Agreed and agreed